robot/web_socket/echo.robot

```diff
@@ -0,0 +1,61 @@
+*** Settings ***
+Documentation     A test suite for examples/web_socket/echo-server
+Library           Collections
+Library           Process
+Library           RequestsLibrary
+Library           WebSocketClient
+
+Suite Setup       Start Servers
+Suite Teardown    Stop Servers
+
+*** Variables ***
+${HTTP_URL}       http://localhost:55501
+${HTTPS_URL}      https://localhost:55502
+${WS_URL}         ws://localhost:55501
+${WSS_URL}        wss://localhost:55502
+${FRAME_COUNT}    ${10}
+${BINARY_PATH}    /path/to/the/server
+${SSL_PATH}       /path/to/the/pem/files
+
+*** Test Cases ***
+Test WebSocket Server
+    [Tags]    WebSocket
+    ${fd}=                  WebSocketClient.Connect  ${WS_URL}
+    WebSocketClient.Send    ${fd}    Hello
+    ${result}=              WebSocketClient.Recv    ${fd}
+    WebSocketClient.Close   ${fd}
+    Should Be Equal         ${result}    Hello
+
+Test WebSocket Over SSL Server
+    [Tags]    WebSocket
+    ${ssl_arg}=             Evaluate      {"cert_reqs": ssl.CERT_NONE}
+    ${fd}=                  WebSocketClient.Connect  ${WSS_URL}   sslopt=${ssl_arg}
+    WebSocketClient.Send    ${fd}    Hello
+    ${result}=              WebSocketClient.Recv    ${fd}
+    WebSocketClient.Close   ${fd}
+    Should Be Equal         ${result}    Hello
+
+*** Keywords ***
+Start Servers
+    ${res1}=    Start Process    ${BINARY_PATH}  -p  55501
+    Set Suite Variable    ${ws_server_process}    ${res1}
+    ${res2}=    Start Process    ${BINARY_PATH}  -p  55502  -k  ${SSL_PATH}/key.pem  -c  ${SSL_PATH}/cert.pem
```

Comment:
There probably isn't a way to avoid a static port?

Reply:
I don't know how. I don't think there's a Robot feature that gives you a currently unused port. But 555xx is in the ephemeral port range (https://en.wikipedia.org/wiki/Ephemeral_port) of IANA/Linux/BSD, so there really shouldn't be any conflicts when running the tests. Especially on CI.